### PR TITLE
Add generate

### PR DESCRIPTION
**Pull request by brettmortensen**
Adds a "generate" GitHub Action that will be used to generate our
library. This can be accessed through the repo's GitHub web UI Actions
-> Generate -> Run workflow. Then you select what branch (master) and
then what kind of version bump (major, minor, patch).

This will then run the workflow generating the library using the
`openapi-generator-cli` npm tool and open up a PR for review.

### CODE REVIEW

File: .github/clean.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require "fileutils"
+
+ALLOW_LIST = [
```
Comment (mcoats13) [inline .github/clean.rb:3]: What about renaming this to be `WHITELIST`?

File: .github/workflows/generate.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          -i https://raw.githubusercontent.com/mxenabled/openapi/master/openapi/mx_platform_api_beta.yml \
+          -g ruby \
+          -c ./openapi/config.yml \
+          -t ./openapi/templates
+    - name: Create branch
+      run: git checkout -b "openapi-generator-${{ steps.bump_version.outputs.version }}"
+    - name: Create commit
+      run: |
+        git config user.name "openapi_generator"
+        git config user.email "openapi_generator@mx.com"
```
Comment (mcoats13) [inline .github/workflows/generate.yml:46]: What do you think about using our team email here?

Comment (brettmortensen) [reply .github/workflows/generate.yml]: Good idea. Just updated.

File: .github/clean.rb (hunk 1 above)
Comment (brettmortensen) [reply .github/clean.rb:3]: I'll dm you.

**Comment by mcoats13** [comment]
:shipit:

**Pull request merged**